src/index/VocabularyMergerImpl.h

```diff
@@ -200,10 +200,20 @@ CPP_template_def(typename C, typename L)(
       external = external || top.isExternal();
     }
     const auto& word = lastTripleComponent_.value();
+
+    // Extract language tag bits from the partial ID to preserve them in the
+    // global ID. The partial ID already contains language tag information in
+    // its lower 20 bits (for VocabIndex types).
+    Id partialId = Id::fromBits(top.id());
```

Comment:
If we make the `top.id()` a fully blown `ID`, then this would be much nicer. (we have to make sure that the blank nodes are correctly handled though).